src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+package controllers;
+
+import models.*;
+import repositories.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+public class BookstoreRestController {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private BookstoreRepository bookstoreRepository;
+    @Autowired
+    private BookstoreOwnerRepository bookstoreOwnerRepository;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+    @Autowired
+    private SaleRepository saleRepository;
+
+
+    //BookstoreOwner REST endpoints
+    @GetMapping("/api/getBookstoreOwner")
+    public BookstoreOwner getBookstoreOwner(@RequestParam(value = "id") long id) {
+        return bookstoreOwnerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstoreOwners")
+    public Iterable<BookstoreOwner> getBookstoreOwner() {
+        return bookstoreOwnerRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstoreOwner")
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "name") String name) {
+        BookstoreOwner bookstoreOwner = new BookstoreOwner(name);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstoreOwner;
+    }
+
+
+    //Bookstore REST endpoints
+    @GetMapping("/api/getBookstore")
+    public Bookstore getBookstore(@RequestParam(value = "id") long id) {
+        return bookstoreRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstores")
+    public Iterable<Bookstore> getBookstores() {
+        return bookstoreRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstore")
+    public Bookstore newBookstore(@RequestParam(value = "name") String name, @RequestParam(value = "bookstoreOwnerID") long bookstoreOwnerID) {
+        BookstoreOwner bookstoreOwner = bookstoreOwnerRepository.findById(bookstoreOwnerID);
+        if (bookstoreOwner == null)
+            return null;
+        Bookstore bookstore = new Bookstore(name);
+        bookstore.setBookstoreOwner(bookstoreOwner);
+        bookstoreOwner.addBookstore(bookstore);
+        bookstoreRepository.save(bookstore);
```

Comment:
Wondering whether this save is necessary, does the bookstoreOwnerRepository.save cascade the save to the book?

Reply:
Yes if you set the cascade type to all

Reply:
bookstoreOwnerRespository.save() does cascade the save for the Bookstore.
Addressed in 774e84af77a3254b78f7c641a9aa0c6d72e5303f